programs/server/config.xml

```diff
@@ -733,6 +733,51 @@
+        <create_graphitemergetree_requires_grant>false</create_graphitemergetree_requires_grant>
+        <create_versionedcollapsingmergetree_requires_grant>false</create_versionedcollapsingmergetree_requires_grant>
+
+        <create_replicatedmergetree_requires_grant>false</create_replicatedmergetree_requires_grant>
+        <create_replicatedcollapsingmergetree_requires_grant>false</create_replicatedcollapsingmergetree_requires_grant>
+        <create_replicatedreplacingmergetree_requires_grant>false</create_replicatedreplacingmergetree_requires_grant>
+        <create_replicatedaggregatingmergetree_requires_grant>false</create_replicatedaggregatingmergetree_requires_grant>
+        <create_replicatedsummingmergetree_requires_grant>false</create_replicatedsummingmergetree_requires_grant>
+        <create_replicatedgraphitemergetree_requires_grant>false</create_replicatedgraphitemergetree_requires_grant>
+        <create_replicatedversionedcollapsingmergetree_requires_grant>false</create_replicatedversionedcollapsingmergetree_requires_grant>
+
+        <create_log_requires_grant>false</create_log_requires_grant>
+        <create_tinylog_requires_grant>false</create_tinylog_requires_grant>
+        <create_stripelog_requires_grant>false</create_stripelog_requires_grant>
+
+        <create_filelog_requires_grant>false</create_filelog_requires_grant>
+        <create_kafka_requires_grant>false</create_kafka_requires_grant>
+        <create_rabbitmq_requires_grant>false</create_rabbitmq_requires_grant>
+        <create_nats_requires_grant>false</create_nats_requires_grant>
+        <create_embeddedrocksdb_requires_grant>false</create_embeddedrocksdb_requires_grant>
+        <create_keepermap_requires_grant>false</create_keepermap_requires_grant>
+
+        <create_null_requires_grant>false</create_null_requires_grant>
+        <create_merge_requires_grant>false</create_merge_requires_grant>
+        <create_buffer_requires_grant>false</create_buffer_requires_grant>
+        <create_memory_requires_grant>false</create_memory_requires_grant>
+        <create_dictionary_requires_grant>false</create_dictionary_requires_grant>
+        <create_set_requires_grant>false</create_set_requires_grant>
+        <create_join_requires_grant>false</create_join_requires_grant>
+        <create_view_requires_grant>false</create_view_requires_grant>
+        <create_materializedview_requires_grant>false</create_materializedview_requires_grant>
+        <create_liveview_requires_grant>false</create_liveview_requires_grant>
+        <create_windowview_requires_grant>false</create_windowview_requires_grant>
+        <create_generaterandom_requires_grant>false</create_generaterandom_requires_grant>
+        <create_executable_requires_grant>false</create_executable_requires_grant>
+        <create_executablepool_requires_grant>false</create_executablepool_requires_grant>
+        <create_fuzzjson_requires_grant>false</create_fuzzjson_requires_grant>
```

Comment:
That seems like too many new settings. Just one new setting like `table_engines_require_grant` should be enough.

Reply:
I think there are some tradeoffs here:
1. If using `table_engines_require_grant`, for example, in order to be grantable to a specific table engine, admins haves to set other old engines as well, means losing control of granularity.
2. the settings in the config file are actually not so necessary. By default would be false according to the code, meaning engine doesn't require grant at all. I list all the setting in the config files just for completeness and illustration purpose. Probably I should just list one, and add some comment for users to reference similar settings, so it won'd be so daunting. What do you think?

Reply:
I agree with you. So currently I would set it to false, and add some comment to warn admins once this feature is enabled (set to true), they have to grant every table engine, otherwise it is by default not granted, what do you think? It seems lengthy for admins. BTW, should some table engines always granted?

Reply:
> BTW, should some table engines always granted?

No, `CREATE USER` should create a user without privileges. This is difficult to change and it's kind of an invariant.

So yes, after this setting will be set to true by default (but it shouldn't happen very soon) it will require some work from admin. But it's not really difficult to grant
```
GRANT TABLE_ENGINE ON * TO username
```